### DISCUSSION TITLE
Turning off type safety

### DISCUSSION THREAD

**Discussion by ekkis** [prisma/prisma#13070] [Q&A]
I have a url that looks like this: `/api/tst?entity=card&userId=3` and code that looks like this:
```js
import { PrismaClient } from "@prisma/client"
const prisma = new PrismaClient();
var ret, db = prisma[req.query.entity]
ret = await db.findMany({ where: req.query })
```
which breaks with:

> [user] Error responding: PrismaClientValidationError: 
> Invalid `prisma.card.findMany()` invocation:
> 
> {
>   where: {
>     customerId: ''
>                 ~~
>   }
> }
> 
> Argument customerId: Got invalid value '' on prisma.findManyCard. Provided String, expected IntFilter or Int:

so now I have to figure out what the user passed in, what the type of the field is and do a conversion.  frankly to me type safety is just an enormous nuisance and I would like to turn it off so it doesn't complicate the living daylights out of my code

can I do it? and if not, what is the recommended approach to this (and please don't tell me to hard-code types.  this is generic code and I don't want to have 1,000 variants on a theme, one for each model such that I can match the bloody types)?

**Appendix I**

I also don't want to have to turn all my entity attributes into strings.  databases like ints for keys

**Comment by trananhkhoi46** [comment]
I checked this Erick,

I think the problem is that "req.query" MIGHT BE not a suitable object that is required by the function "findMany". Inside the req.query, there MIGHT BE some unused parameters (I don't see you pass the customerId field, but somehow that field was inside your req.query object) that you want to take out before passing to the findMany function.

So that my recommendation is
1. You can try to do console.log the object "req.query" to see if it has any dummy fields -> then you can call a code to remove those fields before passing to the findMany function
2. It could be somehow like this
import { PrismaClient } from "@prisma/client"
const prisma = new PrismaClient();
const queryObj = {...req.query};
delete queryObj.customerId;
var ret, db = prisma[req.query.entity]
ret = await db.findMany({ where: queryObj})

**Reply by ekkis** [reply]
my bad.  the url should read: `/api/tst?entity=card&customerId=3`.  the problem, as indicated in the error, is that the `query` object contains strings, therefore `query.customerId` is a string whereas the `where` clause is expecting an integer (because that's how the model attribute is defined).   so that's why I said, I now need to somehow look up the definition for all the items in the query object, and convert to the correct data type e.g. `parseInt(query.customerId)`.  I so hate strong typing.  it's useless and complicates everything

**Reply by trananhkhoi46** [reply]
According to the error, I think that might be not just an error where it needs an integer instead of a string. 
Got invalid value '' on prisma.findManyCard
--> I think the issue is that customerId is an EMPTY string.
if it is a string like '4'   ---> THE PRISMA MIGHT CAN AUTOMATICALLY CONVERT IT TO INTEGER

That is 1 of my concern. could you check that part first? After checking that, if the issue is really what you said (the parameter was expected to be an integer instead of a string). You can try to make a function to loop through all fields of the query object, and try to parse them to integer (if the parse is failed, then we keep that field as a string). something like this

const finalObject = {};
Object.keys(obj).forEach(function(key,index) {
    try {
       finalObject[key] = parseInt(obj[key]);
    } catch(e) {
        finalObject[key] = obj[key];
    }
});

**Reply by ekkis** [reply]
no.  the function expects an integer.  it is strongly typed.  if I use '4' as you suggest it fails.  however, your solution is clever.  I will do that.  thanks a million

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
